multiqc/modules/sequali/sequali.py

```diff
@@ -0,0 +1,499 @@
+        max_length = sorted(max_lengths, reverse=True)[0]
+        self.max_length = max_length
+        min_length = sorted(min_lengths)[0]
+        self.lengths_differ = True
+        self.use_xlog = False
+        if len(min_lengths) == 1 and len(max_lengths) == 1 and min_length == max_length:
+            self.lengths_differ = False
+        if max_length >= 1000:
+            self.use_xlog = True
+        log.info(f"Found {len(self.data)} reports")
+
+        self.sequali_general_stats()
+        self.read_count_plot()
+        self.per_position_quality_plot()
+        self.per_sequence_quality_plot()
+        self.per_position_gc_content_plot()
+        self.per_sequence_gc_content_plot()
+        self.sequence_length_distribution_plot()
+        self.sequence_duplication_levels_plot()
+        self.top_overrepresented_sequences_table()
+        self.adapter_content_plot()
+
+    def sequali_general_stats(self):
+        general_stats = dict()
+        for sample_name, sample_dict in self.data.items():
+            summary = sample_dict["summary"]
+            stats_entry = {
+                "sequali_gc_percentage": (
+                    100 * summary["total_gc_bases"] / max(summary["total_bases"] - summary["total_n_bases"], 1)
+                ),
+                "sequali_mean_sequence_length": summary["mean_length"],
+                "sequali_total_reads": summary["total_reads"],
+                "sequali_duplication_percentage": (1 - sample_dict["duplication_fractions"]["remaining_fraction"])
+                * 100,
+            }
+            general_stats[sample_name] = stats_entry
+        headers = {
+            "sequali_gc_percentage": {
+                "title": "GC %",
+                "description": "Average GC content %",
+                "max": 100,
+                "min": 0,
+                "suffix": "%",
+                "format": "{:.2f}",
+                # GC percentage is not "wrong" in any case. We want a color
+                # scale that clearly distuingishes GC rich vs AT-rich. The
+                # middle 50% should be white, as it is neutral. The
+                # color scale should not include red or green which are coupled
+                # with the 'bad' or 'good' value judgement respectively.
+                # This way we samples with a anomalous GC percentage will
+                # clearly stand out from the others, which is desirable.
+                # The PuOr scale fits these criteria.
+                "scale": "PuOr",
+            },
+            "sequali_mean_sequence_length": {
+                "title": "Mean length",
+                "description": "Geometric mean of all lengths",
+                "min": 0,
+                "suffix": " bp",
+                "format": "{:,.1f}",
+                # Longer reads is considered better for long-read technologies.
+                # Use green to signal that.
+                "scale": "Greens",
```

Comment:
Now both columns are green :) What I meant, is that adjacent columns should be of different colors, so e.g. one is `Blues` and the other is `Greens`.

Reply:
Ah. I misread that. I first had no scales at all, so I fixed that and forgot the other part of the requirement. Sorry.

I choose an arbitrary color scale that is not green. Since it is not green, I had to motivate it a bit more thorough. If you feel the color scale motivation comments are overblown I am happy to remove them.